TESTS/network/wifi/README.md

```diff
@@ -0,0 +1,516 @@
+| 15  | WIFI_SCAN_NULL                          |                            | SHOULD   |
+| 16  | WIFI_SCAN                               |                            | SHOULD   |
+
+Building test binaries
+----------------------
+
+For testing the board and driver, test against the Mbed OS
+master branch to get the most recent, up-to-date test cases and drivers.
+
+To create a build environment:
+
+```.sh
+mbed new wifi_test
+cd wifi_test
+cd mbed-os
+git checkout master
+cd ..
+```
+
+Now build test binaries:
+
+```.sh
+mbed test --compile -t <toolchain> -m <target> -n mbed-os-tests-network-wifi
+```
+
+Running tests
+-------------
+
+Bear in mind that RAAS user and password have to be set in the shell to access it.
+
+```.sh
+mbedgt -g <target>:raas_client:ruka.mbedcloudtesting.com:8000 -n mbed-os-tests-network-wifi -V -v
+```
+
+Test cases for WifiInterface class
+-------------------------------------
+
+### WIFI_CONSTRUCTOR
+
+**Description:**
+
+�Test that constructor of the driver works.
+
+**Preconditions:**
+
+1.  -
+
+**Test steps:**
+
+1.  Construct the driver by calling the constructor.
+
+**Expected result:**
+
+Constructor returns.
+
+### WIFI_CONNECT_NOCREDENTIALS
+
+**Description:**
+
+�Test `WiFiInterface::connect()` without parameters. Don't set parameters with `set_credentials()`
+
+This must be a first test to run after constructing the driver. No credentials should be given for the driver before this test.
+
+**Preconditions:**
+
+1.  Test enviroment is set up as specified in "Test Environment" chapter.
+
+**Test steps:**
+
+1.  Initialise the driver
+2.  Call�`WiFiInterface::connect()`
+3.  `disconnect()`
+
+**Expected result:**
+
+`connect()` call returns�NSAPI_ERROR_PARAMETER or�NSAPI_ERROR_NO_SSID
+
+### WIFI_SET_CREDENTIAL
+
+**Description:**
+
+This test case is to test whether the driver accepts valid credentials and reject ones that are not valid.
+
+**�Preconditions:**
+
+1.  Driver class is initialised.
+
+**Test steps:**
+
+1.  Call `set_credentials()` with various parameters described in "Expected results"
+
+**Expected result:**
+
+| Parameters |                                |                                  | Expected return code                                |
+|------------|--------------------------------|----------------------------------|-----------------------------------------------------|
+| ssid=NULL  | pass=NULL                      | security= NSAPI_SECURITY_NONE    | NSAPI_ERROR_PARAMETER                               |
+| ssid="OK"  | pass=NULL                      | security=NSAPI_SECURITY_WPA_WPA2 | NSAPI_ERROR_PARAMETER                               |
+| ssid="OK"  | pass=NULL                      | security=NSAPI_SECURITY_WEP      | NSAPI_ERROR_PARAMETER                               |
+|            |                                |                                  | because password is missing.                        |
```

Comment:
These two line boxes are not rendering very well.
Just combine the content into one box, like `| NSAPI_ERROR_PARAMETER because password is missing |`

Markdown will then split the lines.

Or you can try `<br />` HTML code to split the lines manually.